In [2]:
import pandas as pd
path = "shopping_trends_updated.csv"
df = pd.read_csv(path)
df.head()

,Customer ID,Age,Gender,Item Purchased,Category,Purchase Amount (USD),Location,Size,Color,Season,Review Rating,Subscription Status,Shipping Type,Discount Applied,Promo Code Used,Previous Purchases,Payment Method,Frequency of Purchases
0,1,55,Male,Blouse,Clothing,53,Kentucky,L,Gray,Winter,3.1,Yes,Express,Yes,Yes,14,Venmo,Fortnightly
1,2,19,Male,Sweater,Clothing,64,Maine,L,Maroon,Winter,3.1,Yes,Express,Yes,Yes,2,Cash,Fortnightly
2,3,50,Male,Jeans,Clothing,73,Massachusetts,S,Maroon,Spring,3.1,Yes,Free Shipping,Yes,Yes,23,Credit Card,Weekly
3,4,21,Male,Sandals,Footwear,90,Rhode Island,M,Maroon,Spring,3.5,Yes,Next Day Air,Yes,Yes,49,PayPal,Weekly
4,5,45,Male,Blouse,Clothing,49,Oregon,M,Turquoise,Spring,2.7,Yes,Free Shipping,Yes,Yes,31,PayPal,Annually


In [3]:
df[['Previous Purchases']].describe()

,Previous Purchases
count,3900.000000
mean,25.351538
std,14.447125
min,1.000000
25%,13.000000
50%,25.000000
75%,38.000000
max,50.000000


### Tugas
### Analisa mean, std, min, max, Q1, Q2, Q3 untuk data Previous Purchases
### Analisa Data Previous Purchases


Setelah dihitung, rata-rata pembelian sebelumnya dari pelanggan itu sekitar 25 kali (25.35 
tepatnya), tapi standar deviasinya cukup tinggi yaitu 14.45. Ini artinya jumlah pembelian 
tiap pelanggan sebenarnya beda-beda jauh, ada yang jarang belanja ada juga yang sering 
banget. Kalau dilihat dari nilai minimumnya, ada pelanggan yang baru belanja 1 kali, 
sedangkan yang paling banyak sampai 50 kali.

Untuk kuartilnya, 25% pelanggan (Q1) punya riwayat pembelian di bawah 13 kali, separuh 
data (Q2/median) ada di angka 25, dan 75% pelanggan (Q3) berada di bawah 38 kali pembelian. 
Karena median-nya (25) hampir sama persis dengan mean (25.35), berarti data ini lumayan 
seimbang, tidak terlalu miring ke kiri atau ke kanan.

In [4]:
df[['Age', 'Purchase Amount (USD)']].corr()

,Age,Purchase Amount (USD)
Age,1.000000,-0.010424
Purchase Amount (USD),-0.010424,1.000000


### Korelasi Age vs Purchase Amount

Angka korelasinya cuma -0.010424, itu udah deket banget sama 0. Artinya umur pelanggan 
nggak ada kaitannya sama besar kecilnya uang yang mereka belanjakan. Yang tua atau yang 
muda, pola belanjanya nggak beda jauh kalau dilihat dari sisi nominal.

In [5]:
df[['Purchase Amount (USD)', 'Review Rating']].corr()

,Purchase Amount (USD),Review Rating
Purchase Amount (USD),1.000000,0.030776
Review Rating,0.030776,1.000000


### Korelasi Purchase Amount vs Review Rating

Sama kayak yang tadi, korelasi Purchase Amount (USD) sama Review Rating juga kecil 
banget, cuma 0.03. Jadi besar kecilnya uang yang dikeluarin pelanggan nggak nentuin 
mereka bakal kasih rating bagus atau jelek. Yang belanja mahal belum tentu puas, 
yang belanja murah juga belum tentu kecewa.

### Kesimpulan

Dari dua hasil korelasi di atas, keliatan jelas kalau Age dan Purchase Amount 
sama-sama nggak punya pengaruh berarti ke variabel lainnya. Kemungkinan ada faktor 
lain yang lebih nentuin perilaku belanja dan kepuasan pelanggan di dataset ini, 
bukan cuma dari umur atau berapa uang yang mereka habiskan.